src/main/java/com/hubspot/imap/utils/formats/ImapDateFormat.java

```diff
@@ -20,6 +23,8 @@
       .toFormatter(Locale.US);
 
   public static final DateTimeFormatter INTERNALDATE_FORMATTER = DateTimeFormatter.ofPattern("d-MMM-yyyy HH:mm:ss Z");
+  private static final DateTimeFormatter INTERNALDATE_FORMATTER_WITH_ZONE = DateTimeFormatter.ofPattern("d-MMM-yyyy HH:mm:ss Z z");
+  public static final Set<DateTimeFormatter> INTERNALDATE_FORMATTERS = ImmutableSet.of(INTERNALDATE_FORMATTER, INTERNALDATE_FORMATTER_WITH_ZONE);
```

Comment:
Should the associated looped-parsing logic exist in this class and not the builder?